extensions/github-stars/src/index.tsx

```diff
@@ -0,0 +1,69 @@
+import {
+  getLocalStorageItem,
+  getPreferenceValues,
+  List,
+  setLocalStorageItem,
+  showToast,
+  ToastStyle,
+} from '@raycast/api'
+import { useEffect, useState } from 'react'
+import fetch from 'node-fetch'
+import { Response } from './response.model'
+import { PackageListItem } from './PackagListItem'
+
+interface State {
+  items?: Response
+  error?: Error
+}
+
+interface ExtensionPreferences {
+  githubUsername: string
+  resultsCount: string
+}
+
+export default function PackageList() {
+  const [state, setState] = useState<State>()
+  const [loading, setLoading] = useState<boolean>(true)
+  const { githubUsername, resultsCount }: ExtensionPreferences =
+    getPreferenceValues()
+
+  useEffect(() => {
+    const fetchPackages = async (): Promise<void> => {
+      const storedItems = await getLocalStorageItem<string>('github-star-items')
+
+      if (storedItems) {
+        setState({ items: JSON.parse(storedItems) as Response })
+      }
+
+      try {
+        const response = await fetch(
+          `https://api.github.com/users/${githubUsername}/starred?per_page=${resultsCount}`,
+        )
+        const json = await response.json()
+        setState({ items: json as Response })
+        setLoading(false)
+        await setLocalStorageItem('github-star-items', JSON.stringify(json))
```

Comment:
If the username is not found there is no validation (and toast)

Reply:
As far as I'm aware, this is not needed since when this extension is first run, this appears: 

![image](https://user-images.githubusercontent.com/64883/140053756-1366a316-a76b-44d9-9449-c82c0b1309ad.png)

Did you not see that? 

Adding validation is trivial, but if that page is always shown, then I don't think it is needed.

Reply:
> As far as I'm aware, this is not needed since when this extension is first run, this appears:
> 
> ![image](https://user-images.githubusercontent.com/64883/140053756-1366a316-a76b-44d9-9449-c82c0b1309ad.png)
> 
> Did you not see that?
> 
> Adding validation is trivial, but if that page is always shown, then I don't think it is needed.

What if the user is deleted, changed or I'm changing the setting manually in the preferences?

Reply:
If the github user is deleted? 🤔  or even if the username is misspelled? I just checked and the extension shows the "no results" page 
![image](https://user-images.githubusercontent.com/64883/140056833-8a0eab6a-d7f4-4b79-9271-ee51ea65ec1e.png)